### PR TITLE
 [13.0] [MIG] product_state

### PR DESCRIPTION
**Pull request by Nikul-Chaudhary**
Product State
=============
This module introduces the state field on the product template and allows a simple product life cycle.

### CODE REVIEW

**Comment by Nikul-Chaudhary** [comment]
@sudhir-erpharbor @aheficent  @zeroheure  can you please review it?

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Cannot install Inventory module, not compatible with stock report. See traceback: 

```
Error:
Odoo Server Error
....
  File "/home/odoo/odoo-13.0/addons/stock/report/report_stock_quantity.py", line 118, in init
    self.env.cr.execute(query)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 163, in wrapper
    return f(self, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/sql_db.py", line 240, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: column reference "state" is ambiguous
LINE 58:         WHEN state = 'done' THEN (now() at time zone 'utc'):...
                      ^
```

Can you check? Thanks

File: product_state/i18n/fr.po [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#. module: product_state
+#: selection:product.template,state:0
+msgid "End of Lifecycle"
+msgstr "Fin de vie"
+
+#. module: product_state
+#: selection:product.template,state:0
+msgid "In Development"
+msgstr "Phase de recherche et de développement"
```
Comment (zeroheure) [inline product_state/i18n/fr.po:30]: This french translation is far too long (this is not a help, but the field title). Further more, it litteraly means "Research and development phase" while the original text is different. I sggest to write "En développement".

**Review by zeroheure — CHANGES_REQUESTED** [review]

**Comment by sudhir-erpharbor** [comment]
> Cannot install Inventory module, not compatible with stock report. See traceback:
> 
> ```
> Error:
> Odoo Server Error
> ....
>   File "/home/odoo/odoo-13.0/addons/stock/report/report_stock_quantity.py", line 118, in init
>     self.env.cr.execute(query)
>   File "/.repo_requirements/odoo/odoo/sql_db.py", line 163, in wrapper
>     return f(self, *args, **kwargs)
>   File "/.repo_requirements/odoo/odoo/sql_db.py", line 240, in execute
>     res = self._obj.execute(query, params)
> psycopg2.ProgrammingError: column reference "state" is ambiguous
> LINE 58:         WHEN state = 'done' THEN (now() at time zone 'utc'):...
>                       ^
> ```
> 
> Can you check? Thanks

It is conflicting with the state field as quants and product both have state field. I think @Nikul-Chaudhary  would need to override the report method and explicitly pass the quants alias in the query like: "quant.state".

**Comment by kos94ok-3D** [comment]
I think need to wait for merge PR: #441

**Comment by emagdalenaC2i** [comment]
@Nikul-Chaudhary Please, see #574

**Comment by rousseldenis** [comment]
@Nikul-Chaudhary Do you mind if we replace by #634 ?

**Comment by rousseldenis** [comment]
Closing this as #634 has been merged

**Pull request closed without merging**